Fixed json tags for volumes

markturansky opened the pull request:
Several "omitempty" tags missing.

Resolves #8755.

markturansky committed:
normalized json tags across volumes

roberthbailey commented:
/cc @bgrant0607 @thockin

bgrant0607 commented on pkg/api/types.go:
```diff
@@ -204,7 +204,7 @@ type VolumeSource struct {
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
 	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
-	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim"`
```
omitempty implies optional. Really we shouldn't have any json tags in this file, but so long as we do, we might as well make the tags correct and consistent. All volume sources should have omitempty here, or just remove the json tags entirely.

markturansky replied:
I'll add omitempty to each.  I believe Tim tried removing all the json tags and failed.

pmorie commented:
Fyi @rootfs
On Mon, May 25, 2015 at 12:07 PM Robert Bailey <notifications@github.com>
wrote:

> /cc @bgrant0607 <https://github.com/bgrant0607> @thockin
> <https://github.com/thockin>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8788#issuecomment-105257463>
> .
>

bgrant0607 commented on pkg/api/v1beta3/types.go:
```diff
@@ -252,7 +252,7 @@ type PersistentVolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs,omitempty" description:"NFS volume resource provisioned by an admin"`
 	// RBD represents a Rados Block Device mount on the host that shares a pod's lifetime
-	RBD *RBDVolumeSource `json:"rbd" description:"rados block volume that will be mounted on the host machine"`
+	RBD *RBDVolumeSource `json:"rbd,omitempty" description:"rados block volume that will be mounted on the host machine"`
```
What about v1beta1 and v1beta2? We should keep them up to date until we remove them.

markturansky replied:
I noticed they were one way while v1beta3 was another.  I thought maybe it was purposeful.  I believe they used to be there.

I'll add 'omitempty' to the other versions, too.

markturansky replied:
nm, i see these were all added by Nikhil but only to v1 and v1beta3. They are all the same now.

bgrant0607 commented:
Thanks, @markturansky.

markturansky committed:
normalized omitempty across volumes for all versions

markturansky commented:
@bgrant0607 I added 'omitempty' to internal types and v1beta1/2, per your request.  

Also, I removed 'omitempty' from PersistentVolumeClaimVolumeSource.ClaimName.  Validation already requires the name, so omitempty makes no sense.

cc @nikhiljindal this continues #8338

thockin commented:
LGTM

bgrant0607 commented:
Thanks. LGTM.

thockin commented:
needs rebase

markturansky committed:
rebased and added 'omitempty' to new volumes

markturansky commented:
Rebased and added 'omitempty' to the couple new volumes that were added to PV.  Waiting on green.

markturansky committed:
added generated deep copy funcs

markturansky commented:
@thockin Travis returned green.
